obspy/io/cybershake/core.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+"""
+Support for reading CyberShake seismograms in ObsPy.
+"""
+
+import struct
+import numpy as np
+from obspy import Trace, Stream
+
+
+NET_CODE = 'CS'  # Assign a network code for CyberShake data
+BAND_INST_CODE = 'MX'  # M indicates mid-period and X indicates synthetic data
+LOCATION_CODE = '00'
+
+
+def _is_cybershake(filename):
+    """
+    Checks whether a file is a CyberShake seismogram or not.
+
+    :type filename: str
+    :param filename: File to be checked.
+    :rtype: bool
+    :return: ``True`` if a CyberShake version 12.10 file.
+    """
+    try:
+        fp_in = open(filename, "rb")
+        header_str = fp_in.read(56)
+        split_point = header_str[0:8].find(b'\x00')
+        version = header_str[0:split_point].decode()
+        assert version == "12.10"
+        return True
+    except Exception:
+        return False
```

Comment:
This needs a (see [our test runner that checks for warnings throwing an exception](https://tests.obspy.org/136240/#7)).. 

```python
finally:
    fp_in.close()
```

..or alternatively, which is easier to read, probably with context manager

```python
    try:
        with open(filename, "rb") as fp_in:
            header_str = fp_in.read(56)
        split_point = header_str[0:8].find(b'\x00')
        version = header_str[0:split_point].decode()
        assert version == "12.10"
        return True
```

As a second thought, this will only work with a filename provided as a string. Ideally, the `_is_...` and `_read_...` function also accept things like open file handles or `io.BytesIO` objects, which gives users more flexibility. You can look at e.g. `obspy/io/sac/core.py` for an example with the `_is_sac` and `_internal_is_sac` routines. (In case of the file type checker routine that involves returning the file pointer to it's original state at the end)
But I'll leave it up to you if you wanna go the distance here.

(ideally we should unify this at some point for all plugins, maybe through a decorator, but nobody got to that yet)

Reply:
Okay I changed this now to allow for file-like objects following the `obspy/io/sac/core.py` example